### PR TITLE
[Enhancement] provide a way to skip building java extensions

### PR DESCRIPTION
**Pull request by dirtysalt**
Fixes #issue

Reason why to skip building java extensions is, some modules take a very long time to build, but we rarely need to use them in development stage.

Like paimon-reader module, it takes a very long time to package all jars into a uber jar.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by kevincai** [comment]
suggest add new commandline option to set the variable, also print the variable in the build parameter summary.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by dirtysalt** [comment]
> suggest add new commandline option to set the variable, also print the variable in the build parameter summary.

Fixed. add a flag `--without-java-ext`

**Review by kevincai — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

File: build.sh [hunk 1]
```diff
@@ -132,6 +134,7 @@ WITH_GCOV=OFF
 WITH_BENCH=OFF
 WITH_CLANG_TIDY=OFF
 USE_STAROS=OFF
+BUILD_JAVA_EXT=ON
```
Comment (Smith-Cruise) [inline build.sh:137]: WITH_JAVA_EXT, similar to the parameters above?

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request merged**